dipy/direction/peaks.py

```diff
@@ -157,8 +157,20 @@ def peak_directions(odf, sphere, relative_peak_threshold=.5,
     return directions, values, indices
 
 
+class PeaksAndMetrics(PeaksAndMetricsDirectionGetter):
+    def __reduce__(self): return _pam_from_attrs, (self.sphere,
+                                                   self.peak_indices,
+                                                   self.peak_values,
+                                                   self.peak_dirs,
+                                                   self.gfa,
+                                                   self.qa,
+                                                   self.shm_coeff,
+                                                   self.B,
+                                                   self.odf)
```

Comment:
Add `self.__class__` as argument to allow sub-classing?   Maybe, it would be better as first argument.